test/user/user_error_cases_test.go

```diff
@@ -0,0 +1,147 @@
+package user
+
+import (
+	"io/ioutil"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hyperledger-labs/orion-server/pkg/constants"
+	"github.com/hyperledger-labs/orion-server/pkg/server/testutils"
+	"github.com/hyperledger-labs/orion-server/pkg/types"
+	"github.com/hyperledger-labs/orion-server/test/setup"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUserTxErrorCases(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(1)
+	setupConfig := &setup.Config{
+		NumberOfServers:     1,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort + 5,
+		BasePeerPort:        pPort + 5,
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+
+	s := c.Servers[0]
+
+	t.Run("creating an user with acl on non-existence database would fail", func(t *testing.T) {
+		aliceCert, _ := testutils.LoadTestCrypto(t, c.GetUserCertDir(), "alice")
+		alice := &types.User{
+			Id:          "alice",
+			Certificate: aliceCert.Raw,
+			Privilege: &types.Privilege{
+				DbPermission: map[string]types.Privilege_Access{
+					"db1": types.Privilege_Read,
+				},
+			},
+		}
+
+		userTx := &types.UserAdministrationTx{
+			UserId:     "admin",
+			TxId:       uuid.New().String(),
+			UserWrites: []*types.UserWrite{{User: alice}},
+		}
+
+		_, err := s.SubmitTransaction(t, constants.PostUserTx, &types.UserAdministrationTxEnvelope{
+			Payload:   userTx,
+			Signature: testutils.SignatureFromTx(t, s.AdminSigner(), userTx),
+		})
+		require.EqualError(t, err, "TxValidation: Flag: INVALID_DATABASE_DOES_NOT_EXIST, Reason: the database [db1] present in the db permission list does not exist in the cluster")
+	})
+
+	t.Run("creating an user with incorrect certificate would fail", func(t *testing.T) {
```

Comment:
Lets also try a valid certificate but from the wrong CA?

Reply:
I have created an issue to handle that separately https://github.com/hyperledger-labs/orion-server/issues/396